src/DistributedTracingManager.ts

```diff
@@ -152,45 +156,33 @@ export class DistributedTracingManager extends BaseExplorer {
         }
 
         if (deviceIds.length === 1) {
-            try {
-                let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
-                await  registry.updateTwin(deviceIds[0], JSON.stringify(twinPatch), twinPatch.etag);
-            } catch (err) {
-                return err.message;
-            }
+            let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+            await registry.updateTwin(deviceIds[0], JSON.stringify(twinPatch), twinPatch.etag);
             return;
         }
 
-        const result = this.scheduleTwinUpdate(twinPatch, iotHubConnectionString, deviceIds);
-        return result;
+        return await this.scheduleTwinUpdate(twinPatch, iotHubConnectionString, deviceIds);
     }
 
-    private scheduleTwinUpdate(twinPatch, iotHubConnectionString: string, deviceIds: string[]) {
-        return new Promise(async (resolve, reject) => {
-            let twinJobId = uuid.v4();
-            let jobClient = iothub.JobClient.fromConnectionString(iotHubConnectionString);
+    private async scheduleTwinUpdate(twinPatch, iotHubConnectionString: string, deviceIds: string[]): Promise<any> {
+        let twinJobId = uuid.v4();
+        let jobClient = iothub.JobClient.fromConnectionString(iotHubConnectionString);
 
-            let queryCondition = this.generateQureyCondition(deviceIds);
-            let startTime = new Date();
-            let maxExecutionTimeInSeconds = 300;
+        let queryCondition = this.generateQureyCondition(deviceIds);
+        let startTime = new Date();
+        let maxExecutionTimeInSeconds = 300;
 
-            try {
-                await jobClient.scheduleTwinUpdate(twinJobId, queryCondition, twinPatch, startTime, maxExecutionTimeInSeconds);
-                const result = await this.monitorJob(twinJobId, jobClient);
-                resolve("\nDetailed information are shown as below:\n" + JSON.stringify(result, null, 2));
-            } catch (err) {
-                reject("Could not monitor distributed tracing setting update job: " + err);
-            }
-        });
+        await jobClient.scheduleTwinUpdate(twinJobId, queryCondition, twinPatch, startTime, maxExecutionTimeInSeconds);
+        return await this.monitorJob(twinJobId, jobClient);
```

Comment:
No need for `return await `, just return

Reply:
fixed